proposals/2024-09-01-Tags-Improvement.md

```diff
@@ -0,0 +1,154 @@
+
+```yaml
+tags:
+- name: deprecated
+  kind: internal
+  summary: Deprecated
+  description: This operation has been deprecated and will be removed in the future. Avoid using items with this tag.
+- name: shop
+  kind: nav
+  summary: Order Online
+  description: Operations relating to the retail operations behind the [online shopping site](https://example.com/shopping).
+- name: products
+  kind: nav
+  parent: shop
+  summary: Products
+  description: View and manage the product catalog.
+- name: orders
+  kind: nav
+  parent: shop
+  summary: Online Orders
+  description: Place, fulfil and invoice orders for the online shop.
+
+```
+
+Rather than making an allowed list of kinds, we will instead leave that open for user extension and keep a list of the recommended/expected types in a registry and evolve that as conventions emerge.
+
+## Detailed design
+
+The following section is an updated specification section, for the top-level tags object only (no other changes are needed):
+
+---
+
+#### Tag Object
+
+Adds metadata to a single tag that is used by the [Operation Object](#operation-object).
+Each tag used in the Operation Object instances MAY have a Tag Object defined.
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="tagName"></a>name | `string` | **REQUIRED**. The name of the tag. Use this value in the `tags` array of an Operation.
+<a name="tagSummary"></a>summary | `string` | A short summary of the tag, used for display purposes.
+<a name="tagDescription"></a>description | `string` | A description for the tag. [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.
+<a name="tagExternalDocs"></a>externalDocs | [External Documentation Object](#external-documentation-object) | Additional external documentation for this tag.
+<a name="tagParent"></a>parent | `string` | The `name` of a tag that this tags is nested under. The named tag MUST exist in the API description, and circular references between parent and child tags MUST NOT be used.
+<a name="tagKind"></a>kind | `string` | A machine-readable string to categorize what sort of tag it is. Common uses are `nav` for Navigation, `badge` for badges, `internal` for internal APIs, but any string value can be used. A registry of known values is available.
+
+This object MAY be extended with [Specification Extensions](#specification-extensions).
+
+##### Tag Object Example
+
+```json
+{
+	"name": "account-updates",
+    "summary": "Account Updates",
+	"description": "Account update operations",
+    "type": "nav"
+},
+{
+	"name": "partner",
+    "summary": "Partner",
+	"description": "Operations available to the partners network",
+    "parent": "external",
+    "type": "audience"
+}
+```
+
+```yaml
+- name: account-updates
+  summary: Account Updates
+  description: Account update operations
+  type: nav
+
+- name: partner
+  summary: Partner
+  description: Operations available to the partners network
+  parent: external
```

Comment:
Can we add the `external` tag?

Reply:
@ralfhandl I view this text as just giving us the general idea of how the proposal would be implemented.  We can nail down the spec wording and examples when the spec changes are made.

(we should probably figure out what the balance is between the proposals and the specification text, idk, maybe folks want to get the wording right here, in which case I'm happy to defer to that approach)

Reply:
Depending on how long this proposal will sit here without being incorporated into a specification version I'd like to avoid the impression that the proposal allows/advertises the use of dangling "parent" values by adding just one more tag object to the example and make it self-contained.

Reply:
@ralfhandl I see- that makes sense.